### ISSUE TITLE
MeiliSearchApiException throw due to `Json deserialize error`

### ISSUE THREAD

**Issue by Andoctorey** [meilisearch/meilisearch-java#358]
Filter doesn't work for me same as other attributes that requires array. Code example:
```
val searchRequest = SearchRequest(query).also {
     it.matches = true
     it.attributesToRetrieve = null
     it.limit = limit
     it.offset = offset
     it.filter = arrayOf("title = media")
}
 index.search(searchRequest)
```
This returns:
>  MeiliSearchApiException{error=APIError{message='Was expecting an operation `=`, `!=`, `>=`, `>`, `<=`, `<`, `TO` or `_geoRadius` at `[Ljava.lang.String;@8f8841a`.
    1:28 [Ljava.lang.String;@8f8841a', code='invalid_filter', type='invalid_request', link='https://docs.meilisearch.com/errors#invalid_filter'}}
        at com.meilisearch.sdk.MeiliSearchHttpRequest.post(MeiliSearchHttpRequest.java:90)
        at com.meilisearch.sdk.Search.rawSearch(Search.java:93)
        at com.meilisearch.sdk.Search.search(Search.java:166)
        at com.meilisearch.sdk.Index.search(Index.java:284)

I checked in debug what library sends:
> api = /indexes/global/search
> body = {"q":"","offset":0,"limit":20,"cropLength":200,"matches":true,"filter":"[Ljava.lang.String;@169dc22"}

It looks like there is issue with json serializer in your library, please fix.

**Comment by alallema** [comment]
Hi @Andoctorey,
You can write your request this way:
```
      SearchRequest searchRequest =
        new SearchRequest(query)
          .setMatches(true)
          .setAttributesToRetrieve(null)
          .setLimit(limit)
          .setOffset(offset)
          .setFilter(new String[] {"title = media"});
      index.search(searchRequest);
```
It should work.

**Comment by Andoctorey** [comment]
@alallema Do you mean don't use Kotlin and use Java?

**Comment by Andoctorey** [comment]
Just tried pure Java calls - have same error.

**Comment by TomBastable** [comment]
I'm having almost exactly the same issue here. I've spoken to Amelie via the Meili Slack - we're going to try upgrading the server to 0.26 and the library to 0.7.1 to see if that helps at all. Neither Kotlin or Java yields different results.

**Comment by cuongdxk57** [comment]
I get the same error message when using Android Studio, but it run correctly on Eclipse  
MeiliSearchApiException{error=APIError{message='Json deserialize error: invalid type: string "[Ljava.lang.String;@2eb4bb4", expected a sequence at line 1 column 87', code='bad_request', type='invalid_request', link='https://docs.meilisearch.com/errors#bad_request'}

**Comment by TomBastable** [comment]
We removed this from our stack for now and just use Ktor to query the server now.

**Comment by Andoctorey** [comment]
@TomBastable Is this supported in 0.7.2?

**Comment by mayangzz** [comment]
hello, I encounter the similar issue using Kotlin. Is there any development ?

**Comment by alallema** [comment]
Hi @mayangzz,
Not yet, but the SDK is being rewritten I hope it will be available soon!

**Comment by MoustafaElsaghier** [comment]
@alallema I'm trying to use it in kotlin 

                 val client = Client(Config("URL",
                    "API_KEY"))

                val index: Index = client.index("API_INDEX")

                // Meilisearch is typo-tolerant:
                val results: SearchResult = index.search(query)
                Timber.d("RESULT => $results")

but, I'm getting error messages in catch block

> Meilisearch ApiException: {Error=APIError: {message='Json deserialize error: unknown field `matches`, expected one of `q`, `offset`, `limit`, `page`, `hitsPerPage`, `attributesToRetrieve`, `attributesToCrop`, `cropLength`, `attributesToHighlight`, `showMatchesPosition`, `filter`, `sort`, `facets`, `highlightPreTag`, `highlightPostTag`, `cropMarker`, `matchingStrategy` at line 1 column 67', code='bad_request', type='invalid_request', link='https://docs.meilisearch.com/errors#bad_request'}}


so, is there any suggestion why that happens, as far for me I'm doing the search query but, it's not happening so, can you let me am I missing anything from my side?

**Comment by alallema** [comment]
Hi @MoustafaElsaghier,
As explained in this issue #494, the latest version of this SDK is currently only compatible with Meilisearch v0.27 at most.

**Comment by MoustafaElsaghier** [comment]
so, what should I do?
should I change the version on my side or what?
this is my current version 
`    
implementation 'com.meilisearch.sdk:meilisearch-java:0.8.0'
`

**Comment by alallema** [comment]
This issue should finally be resolved as the SDK has been updated to version v0.30.0 of Meilisearch

**Comment by coyzeng** [comment]
> This issue should finally be resolved as the SDK has been updated to version v0.30.0 of Meilisearch

v0.30.0? The latest version is 0.15.0.
